lib/ui/painting/codec.cc

```diff
@@ -79,10 +84,79 @@ static sk_sp<SkImage> DecodeImage(fml::WeakPtr<GrContext> context,
   }
 }
 
+static bool needsResize(const double currentWidth,
+                        const double currentHeight,
+                        const double targetWidth,
+                        const double targetHeight) {
+  if (fuzzyEquals(currentWidth, kDoNotResizeImage) ||
+      fuzzyEquals(currentHeight, kDoNotResizeImage)) {
```

Comment:
Instead of doing `fuzzyEquals`, I think checking `targetWidth < 0` seems to be a more robust check.

Reply:
I wanted to be consistent with the usage of `kDoNotResizeImage`, that way somebody in the future could trace the usage of it and replace it with `std::optional` if/once we have `c++17`.

Reply:
Sure, see my new comments on integers vs doubles. Maybe just make it integer -1 and check `!= kDoNotResizeImage` here. BTW, do you think that `kNoResizeDimension` is a better name?  `kDoNotResizeImage` looks like a boolean to me.